rust/core-lib/src/styles.rs

```diff
@@ -220,4 +240,159 @@ impl ThemeStyleMap {
         self.styles.push(style.clone());
         result
     }
+
+    /// Delete key from the themes map.
+    pub(crate) fn remove_theme(&mut self, path: &Path) -> Option<String> {
+        if invalid_theme_file(path) {
+            return None;
+        }
+        let theme_name = path.file_stem().and_then(OsStr::to_str)?;
+        self.themes.themes.remove(theme_name);
+        self.state.remove(path);
+
+        //Delete dump file
+        let dump_p = self.get_dump_path(theme_name)?;
+        if dump_p.exists() {
+            let _ = fs::remove_file(dump_p);
+        }
+
+        Some(theme_name.to_string())
+    }
+
+    /// Load all themes inside the given directory.
+    pub(crate) fn load_theme_dir(&mut self) {
+        if let Some(themes_dir) = self.themes_dir.clone() {
+            match ThemeSet::discover_theme_paths(themes_dir) {
+                Ok(themes) => {
+                    self.caching_enabled = self.caching_enabled 
+                        && self.init_cache_dir();
+                    for theme_p in themes.iter() {
+                        //This is called at startup
+                        //We need to filter out dumps
+                        //whose theme files were changed
+                        //when editor was closed.
+                        self.load_theme(theme_p);
+                    }
+                },
+                Err(e) => eprintln!("Error while loading themes dir: {:?}",e)
+            }
+        }
+    }
+
+    fn load_theme(&mut self, theme_p: &Path) {
+        match self.try_load_from_dump(theme_p) {
+            Some((k, v)) => {
+                self.insert_to_map(k, v, theme_p);
+            },
+            None => {
+                let _ = self.insert_theme(theme_p);
+            }
+        }
+    }
+
+    // A wrapper around `from_dump_file`
+    // to validate the state of dump file.
+    fn try_load_from_dump(&self, theme_p: &Path)
+        -> Option<(String, Theme)>
+    {
+        if !self.caching_enabled { return None; }
+
+        let theme_name = theme_p.file_stem()
+            .and_then(OsStr::to_str)?;
+
+        let dump_p = self.get_dump_path(theme_name)?;
+
+        if !&dump_p.exists() { return None; }
+
+        let mod_t = fs::metadata(&dump_p)
+            .and_then(|md| md.modified()).ok()?;
+        let mod_t_org = fs::metadata(theme_p)
+            .and_then(|md| md.modified()).ok()?;
+
+        match mod_t.duration_since(mod_t_org) {
```

Comment:
Actually, the `duration_since` returns `ok(time_difference)` if the `self` time is greater than the argument. Otherwise this returns an error. As I'm not using the `time_difference` here so I must replace it with an if-else branch.

Reply:
I think `if let ... {} else {}` or `unwrap_or_else` would be nice rusty ways to do this